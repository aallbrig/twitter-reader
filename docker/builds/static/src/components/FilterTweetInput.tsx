import * as React from 'react';
import {
    Well,
    FormGroup,
    ControlLabel,
    FormControl,
    OverlayTrigger,
    Popover,
    Badge
} from 'react-bootstrap';

export interface Props {
    onKeyDown: (value: string) => void;
}

const filterTweetsTooltip = (
    <Popover id="filter-input-tooltip">
        Start typing in words to match text in the tweets.
        <br />
        If there is a match then the matched text will be highlighted.
        All other tweets without matches will be hidden.
    </Popover>
);

export const FilterTweetInput: React.SFC<Props> = ({ onKeyDown }) => (
    <Well bsSize="large" style={{ position: 'relative' }}>
        <div style={{ position: 'absolute', top: 10, right: 10 }}>
            <OverlayTrigger
                placement="left"
                overlay={filterTweetsTooltip}
                trigger={['click', 'hover', 'focus']}
                delayHide={4000}
            >
                <Badge>?</Badge>
            </OverlayTrigger>
        </div>
        <FormGroup>
            <ControlLabel>
                Tweet Content Filter & Highlight
            </ControlLabel>
            <FormControl
                type="text"
                placeholder="Highlight text in tweet content"
                onChange={(e) => onKeyDown((e.target as any).value)}
            />
        </FormGroup>
    </Well>
);

export default FilterTweetInput;
