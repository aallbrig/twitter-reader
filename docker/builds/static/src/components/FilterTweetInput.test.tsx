import * as React from 'react';
import FilterTweetInput from './FilterTweetInput';
import { create } from 'react-test-renderer';
import { noop } from 'lodash';

it('FilterTweetInput renders correctly', () => {
  const tree = create(
    <FilterTweetInput onKeyDown={noop} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});