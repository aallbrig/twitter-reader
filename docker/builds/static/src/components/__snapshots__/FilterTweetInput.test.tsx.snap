// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTweetInput renders correctly 1`] = `
<div
  className="well well-lg"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "position": "absolute",
        "right": 10,
        "top": 10,
      }
    }
  >
    <span
      className="badge"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      ?
    </span>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor={undefined}
    >
      Tweet Content Filter & Highlight
    </label>
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="Highlight text in tweet content"
      type="text"
    />
  </div>
</div>
`;
